test/pools/PoolPriceOracle.test.ts

```diff
@@ -0,0 +1,291 @@
+          const expectedAccLogBptPrice = previousSample.accLogBptPrice.add(bn(newLogBptPrice).mul(actualElapsed));
+          expect(newSample.accLogBptPrice).to.be.equal(expectedAccLogBptPrice);
+
+          const lastLogInvariant = useLastInvariant ? previousSample.lastLogInvariant : newLogInvariant;
+          expect(newSample.lastLogInvariant).to.be.equal(lastLogInvariant);
+
+          const expectedAccLogInvariant = previousSample.accLogInvariant.add(bn(lastLogInvariant).mul(actualElapsed));
+          expect(newSample.accLogInvariant).to.be.equal(expectedAccLogInvariant);
+        });
+      };
+
+      context('when there was no sample in the given index', () => {
+        const index = 0;
+        // Some elapsed time must exist the first time the oracle is updated
+        itCreatesAnotherSample(index, MINUTE * 60);
+      });
+
+      context('when there was a sample in the given index', () => {
+        context('when the next sample does not complete the buffer', () => {
+          const index = 506;
+
+          sharedBeforeEach('create a sample', async () => {
+            await oracle.mockSample(index, {
+              lastLogPairPrice: 1,
+              accLogPairPrice: 10,
+              lastLogBptPrice: 2,
+              accLogBptPrice: 20,
+              lastLogInvariant: 3,
+              accLogInvariant: 30,
+              timestamp: await currentTimestamp(),
+            });
+          });
+
+          context('when the current timestamp is the same as the initial timestamp of the current sample', () => {
+            itUpdatesTheExistingSample(index, 0);
+          });
+
+          context('when the current timestamp is greater than the initial timestamp by less than 2 minutes', () => {
+            itUpdatesTheExistingSample(index, MINUTE);
+          });
+
+          context('when the current timestamp is greater than the initial timestamp by more than 2 minutes', () => {
+            itCreatesAnotherSample(index, MINUTE * 3);
+          });
+        });
+
+        context('when the next sample does not complete the buffer', () => {
+          const index = MAX_BUFFER_SIZE - 1;
+
+          sharedBeforeEach('create a sample', async () => {
+            await oracle.mockSample(index, {
+              lastLogPairPrice: 1,
+              accLogPairPrice: 10,
+              lastLogBptPrice: 2,
+              accLogBptPrice: 20,
+              lastLogInvariant: 3,
+              accLogInvariant: 30,
+              timestamp: await currentTimestamp(),
+            });
+          });
+
+          context('when the current timestamp is greater than the initial timestamp by less than 2 minutes', () => {
+            itUpdatesTheExistingSample(index, MINUTE);
+          });
+
+          context('when the current timestamp is greater than the initial timestamp by 2048 minutes', () => {
+            itUpdatesTheExistingSample(index, MINUTE * 3, true);
+          });
+        });
```

Comment:
I think we're missing this scenario here:

          context('when the current timestamp is greater than the initial timestamp by more than 2 minutes', () => {
            itCreatesAnotherSample(index, MINUTE * 3);
          });